xskillscore/tests/test_deterministic.py

```diff
@@ -212,25 +301,93 @@ def test_correlation_metrics_xr_dask(
-@pytest.mark.parametrize("skipna", [True, False])
-@pytest.mark.parametrize("has_nan", [True, False])
 def test_distance_metrics_xr_dask(
-    a_dask, b_dask, dim, weight_bool, weights_dask, metrics, skipna, has_nan
+    a_dask, b_dask, dim, weight_bool, weights_dask, metrics
 ):
     """Test whether distance metrics for xarray functions can be lazy when
-    chunked by using dask and give same results."""
+    chunked by using dask and give same results as np array."""
     a = a_dask.copy()
-    if has_nan:
-        a = a.load()
-        a[0] = np.nan
-        a = a.chunk()
     b = b_dask.copy()
     weights = weights_dask.copy()
     # unpack metrics
     metric, _metric = metrics
     # Generates subsetted weights to pass in as arg to main function and for
     # the numpy testing.
     _weights = adjust_weights(dim, weight_bool, weights)
+    if _weights is not None:
+        _weights = _weights.load()
+    if metric is median_absolute_error:
+        actual = metric(a, b, dim)
+    else:
+        actual = metric(a, b, dim, weights=_weights)
+    # check that chunks for chunk inputs
+    assert actual.chunks is not None
+    if metric is median_absolute_error:
+        expected = metric(a.load(), b.load(), dim)
+    else:
+        expected = metric(a.load(), b.load(), dim, weights=_weights)
+    assert expected.chunks is None
+    assert_allclose(actual.compute(), expected)
+
+
+@pytest.mark.parametrize("metrics", correlation_metrics)
+@pytest.mark.parametrize("dim", AXES)
+@pytest.mark.parametrize("weight_bool", [True, False])
+@pytest.mark.parametrize("skipna", [True, False])
+@pytest.mark.parametrize("has_nan", [True])
+def test_correlation_metrics_xr_dask_nan(
+    a_dask_nan, b_dask_nan, dim, weight_bool, weights_dask, metrics, skipna, has_nan
+):
+    """Test whether correlation metric for xarray functions can be lazy when
+    chunked by using dask and give same results as np array with nans."""
+    a = a_dask_nan
+    b = b_dask_nan
+    weights = weights_dask
+    # unpack metrics
+    metric, _metric = metrics
+    # Only apply over time dimension for effective p value.
+    if (dim != "time") and (metric in temporal_only_metrics):
+        dim = "time"
+    # Generates subsetted weights to pass in as arg to main function and for
+    # the numpy testing.
+    _weights = adjust_weights(dim, weight_bool, weights)
+
+    if metric in temporal_only_metrics:
+        actual = metric(a, b, dim, skipna=skipna)
+    else:
+        actual = metric(a, b, dim, weights=_weights, skipna=skipna)
+    # check that chunks for chunk inputs
+    assert actual.chunks is not None
+
+    if _weights is not None:
+        _weights = _weights.load()
+
+    if metric in temporal_only_metrics:
+        expected = metric(a.load(), b.load(), dim, skipna=skipna)
+    else:
+        expected = metric(a.load(), b.load(), dim, _weights, skipna=skipna)
+    assert expected.chunks is None
+    assert_allclose(actual.compute(), expected)
```

Comment:
We don't even need all the computes and loads. Assert triggers that on its own...